### PR TITLE
[IMP] stock: Add Check Availability button to picking list view

### PR DESCRIPTION
**Pull request by yhu-odoo**
Add Check Availability button to the control panel of picking list view.

Task 2361283

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock/models/stock_picking.py [hunk 1]
```diff
@@ -700,6 +700,8 @@ def action_assign(self):
         also impact the state of the picking as it is computed based on move's states.
         @return: True
         """
+        if self.filtered(lambda picking: picking.state == 'done'):
+            raise UserError(_('You cannot reserve a picking that has been set to "Done".'))
         self.filtered(lambda picking: picking.state == 'draft').action_confirm()
```
Comment (ryv-odoo) [inline addons/stock/models/stock_picking.py:705]: Do we want to auto-confirm all draft picking in the same time ? (I do see any
It is maybe return a error for draft as for the "done" state ? Or do nothing for both state and don't raise a error ?

Comment (yhu-odoo) [reply addons/stock/models/stock_picking.py:705]: @ryv-odoo I'm not sure. Why you think confirm all draft picking can be dangerous?

Comment (ryv-odoo) [reply addons/stock/models/stock_picking.py:705]: After some discussion with arm about that: it is Ok to do that. We can let it like that.

Comment (ryv-odoo) [reply addons/stock/models/stock_picking.py:705]: Maybe you can merge your commit, it is the same topic.

Comment (yhu-odoo) [reply addons/stock/models/stock_picking.py:705]: done

**Review by ryv-odoo — APPROVED** [review]

File: addons/stock/models/stock_picking.py [hunk 2]
```diff
@@ -700,6 +700,8 @@ def action_assign(self):
         also impact the state of the picking as it is computed based on move's states.
         @return: True
         """
+        if self.filtered(lambda picking: picking.state == 'done'):
```
Comment (amoyaux) [inline addons/stock/models/stock_picking.py:703]: You could just filter out done picking.

Comment (yhu-odoo) [reply addons/stock/models/stock_picking.py]: ok, removed

File: addons/stock/models/stock_picking.py [hunk 3]
```diff
@@ -710,7 +710,6 @@ def action_assign(self):
         package_level_done.write({'is_done': False})
         moves._action_assign()
         package_level_done.write({'is_done': True})
-
```
Comment (amoyaux) [inline addons/stock/models/stock_picking.py:713]: small useless diff and we are good to go

Comment (yhu-odoo) [reply addons/stock/models/stock_picking.py]: oops.... removed

**Comment by Whenrow** [comment]
robodoo r+

**Pull request closed without merging**